src/ngraph_builder.cc

```diff
@@ -119,7 +119,71 @@ static tf::Status TranslateUnaryOp(tf::Node* op, Builder::OpMap& ng_op_map) {
   return TranslateUnaryOp(op, ng_op_map, [](std::shared_ptr<ng::Node> n) { return make_shared<T>(n); });
 }
 
-// Helper for Builder::TranslateGraph (elementwise binops)
+// Helper function to translate a binary op
+// Parameters:
+//
+//    tf::Node* op               - TF op being translated. Must have only two inputs.
+//    Builder::OpMap& ng_op_map  - The TF-to-nGraph op map.
+//
+//    std::function<std::shared_ptr<ng::Node>(std::shared_ptr<ng::Node>, std::shared_ptr<ng::Node>)>
+//      build_graph              - Function to construct the graph implementing
+//                                 the binary op, given the 2 ng_inputs to the binaryop
+//                                 as an argument.
+// Example Usage:
+// 
+// if (op->type_string() == "SquaredDifference") {
+//      TF_RETURN_IF_ERROR(TranslateBinaryOp(op, ng_op_map,
+//         [](std::shared_ptr<ng::Node> ng_input1, std::shared_ptr<ng::Node> ng_input2) {
+//           auto ng_diff = std::make_shared<ng::op::Subtract>(input1, input2);
+//           return std::make_shared<ng::op::Multiply>(ng_diff,ng_diff);
+//         }));
+//    }
+//
+
+static tf::Status TranslateBinaryOp(
+    tf::Node* op, Builder::OpMap& ng_op_map,
+    std::function<std::shared_ptr<ng::Node>(std::shared_ptr<ng::Node>, std::shared_ptr<ng::Node>)>
```

Comment:
It's a a good use of `lambda` (and you followed the `TranslateUnaryOp(...)` example which is good). Just the name of the variable is a bit confusing. Instead of calling it `build_graph` consider some other name such as `create_binary_op`. 

Also change the name of the variable that holds the lambda in the `TranslateUnaryOp(...)` as well to make it consistent.

Reply:
Agreed. Have changed as suggested.

Reply:
Looks like you haven't pushed the changes?